app/assets/src/components/common/MetadataCSVLocationsMenu.jsx

```diff
@@ -0,0 +1,192 @@
+    } = this.props;
+
+    const newValue = (find({ [NAME_COLUMN]: sample }, metadata.rows) || {})[
+      metadataType.name
+    ];
+    const newMetadata = metadata;
+    const allWarnings = {};
+
+    // Set all the rows to the newValue
+    newMetadata.rows.forEach(row => {
+      const { result, warning } = processLocationSelection(
+        newValue,
+        isRowHuman(row)
+      );
+      row[metadataType.name] = result;
+      if (warning) allWarnings[row[NAME_COLUMN]] = warning;
+    });
+
+    onMetadataChange({
+      metadata: newMetadata,
+    });
+    onCSVLocationWarningsChange(allWarnings);
+    this.setState({ applyToAllSample: null });
+  };
+
+  // Populate the data table cells
+  getManualInputData = () => {
+    const {
+      CSVLocationWarnings,
+      metadata,
+      metadataType,
+      onMetadataChange,
+    } = this.props;
+
+    return metadata.rows.map((row, rowIndex) => {
+      const sampleName = row[NAME_COLUMN];
+
+      const onChange = (key, value) => {
+        const newMetadata = metadata;
+        newMetadata.rows[rowIndex][metadataType.name] = value;
+
+        onMetadataChange({
+          metadata: newMetadata,
+        });
+        this.setState({ applyToAllSample: sampleName });
+        logAnalyticsEvent("MetadataManualInput_input_changed", {
+          key,
+          value,
+          sampleName,
+        });
+      };
+
+      return {
+        [NAME_COLUMN]: (
+          <div className={cs.sampleName} key={NAME_COLUMN}>
+            {sampleName}
+          </div>
+        ),
+        [metadataType.key]: (
+          <div>
+            <MetadataInput
+              key={metadataType.key}
+              className={cs.input}
+              value={row[metadataType.name]}
+              metadataType={metadataType}
+              onChange={onChange}
+              withinModal={true}
+              isHuman={isRowHuman(row)}
+              warning={CSVLocationWarnings[sampleName]}
+            />
+            {this.renderApplyToAll(sampleName)}
+          </div>
+        ),
+      };
+    });
+  };
+
+  render() {
+    const { metadata, metadataType } = this.props;
+
+    if (!(metadata && metadata.rows)) return null;
+    return (
+      <React.Fragment>
+        <div className={cs.instructions}>
+          <div className={cs.title}>Location Matches</div>
+          <div className={cs.subtitle}>
+            We automatically searched for location matches. Please double check
+            and correct any errors.
+          </div>
+        </div>
+        <DataTable
+          className={cs.inputTable}
+          headers={{
+            [NAME_COLUMN]: NAME_COLUMN,
+            [metadataType.key]: metadataType.name,
+          }}
+          columns={[NAME_COLUMN, metadataType.key]}
+          data={this.getManualInputData()}
+        />
+      </React.Fragment>
+    );
+  }
+}
+
+MetadataCSVLocationsMenu.propTypes = {
+  CSVLocationWarnings: PropTypes.object,
+  metadata: PropTypes.object,
```

Comment:
Can you do `PropTypes.shape` here to document better what the structure of this object is? For readability.

Reply:
👍 done